[cliftonchee] iP

cliftonchee opened the pull request:
TheCount
> “AH-AH-AH! One!"

TheCount helps to count and keep track of your tasks, it's

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/cliftonchee/ip/releases).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public static void main(String[] args) {
        new TheCount().run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

cliftonchee committed:
Add standard replies

cliftonchee committed:
Add standard replies

cliftonchee committed:
Rename chatbot and check for exit condition

cliftonchee committed:
Change speech patterns for chatbot

cliftonchee committed:
Added add and list functionality

cliftonchee committed:
Added Task and to mark

cliftonchee committed:
Added logic for mark and unmarkTask

cliftonchee committed:
Added different types of tasks

cliftonchee committed:
Added Text UI testing

cliftonchee committed:
Added exception handling

cliftonchee committed:
Delete function and error handling

cliftonchee committed:
Refactored and added enums

cliftonchee committed:
Refactoring of code

cliftonchee committed:
Level-7: Save

cliftonchee committed:
Merge branch 'branch-Level-7'

cliftonchee committed:
Level 8: Dates and Times

cliftonchee committed:
Merge branch 'branch-Level-8'

cliftonchee committed:
A-MoreOOP

cliftonchee committed:
A-Packages

cliftonchee committed:
Merge remote-tracking branch 'origin/add-gradle-support'

cliftonchee committed:
Add Gradle Support

cliftonchee committed:
Add JUnit Testing

cliftonchee committed:
Add JavaDocs

cliftonchee committed:
Add Java Coding Standards

cliftonchee committed:
Add find functionality

cliftonchee committed:
Fix CheckStyle errors

cliftonchee committed:
Merge branch 'branch-A-CheckStyle'

cliftonchee committed:
Add skeleton for GUI

cliftonchee committed:
Fix FXML errors

tanqinyong commented on src/main/java/thecount/exception/TheCountException.java:
Great to see you making your own exceptions!

tanqinyong commented on src/main/java/thecount/parser/Parser.java:
```diff
@@ -0,0 +1,233 @@
+    private Storage loader;
+
+    /**
+     * Represents the types of commands.
```
I like use of enums!

AwJoey replied:
Agreed! 💯

tanqinyong commented on src/main/java/thecount/parser/Parser.java:
I like how you abstracted out many different functions! This class is really complete. LGTM

cliftonchee replied:
Thanks!

tanqinyong reviewed (commented):
Overall, I found your code to be pretty complete and well-designed. It was easy to read and neat.

cliftonchee committed:
Add GUI

cliftonchee committed:
Merge branch 'branch-Level-10'

cliftonchee committed:
Added assertions, leaving assumptions unchecked and leading to runtime errors.

They are crucial to validate assumptions and catch potential issues early in development.

Added assertions to codebase.

They help to identify and resolve issues before they impact the system's behavior.

Java assertions help to enhance code maintainability and facilitating debugging efforts.

ShanylOng commented:
Plenty of personalisation in this chatbot. It's very well documented as well.

cliftonchee committed:
Refactor and enhance code quality

* Extracted repetitive logic into reusable methods to promote code reuse.
* Applied consistent formatting and indentation across the codebase.
* Encapsulated complex logic into separate classes for better modularity.
* Ensured adherence to Java coding standards and best practices.

cliftonchee committed:
Merge pull request #1 from cliftonchee/branch-A-Assertions

Branch assertions

cliftonchee committed:
Merge branch 'master' into branch-A-CodeQuality

cliftonchee committed:
Merge branch 'branch-A-CodeQuality'

cliftonchee committed:
Add tagging feature

Allows users to tag tasks with custom labels, enhancing task management within the app.

Changes included in this commit:
* Modified the user interface to accept commands for tagging tasks.
* Updated the  class to support tagging of tasks with custom labels.
* Added error handling for empty tags and invalid task numbers during tagging.

cliftonchee committed:
Test CI

cliftonchee committed:
Change filepath for runtest.sh

cliftonchee committed:
Remove text-based UI test

cliftonchee committed:
Removed text-based UI test

cliftonchee committed:
Merge branch 'master' of https://github.com/cliftonchee/ip

cliftonchee committed:
Refactor of Parser

cliftonchee committed:
Remove unnecessary comments

cliftonchee committed:
Add JavaDoc

AwJoey commented on src/main/java/Duke.java:
I like your package categorization, it's really neat! 💯

AwJoey commented on src/main/java/thecount/task/Deadline.java:
```diff
@@ -0,0 +1,106 @@
+     * @return The type of the task.
+     */
+    @Override
+    public String getType() {
```
Like that you implemented this method, making the type value easy to change if needed in future 💯

AwJoey commented on src/main/java/thecount/ui/Reply.java:
```diff
@@ -0,0 +1,24 @@
+/**
+ * Represents a reply message to be displayed to the user.
+ */
+public class Reply {
```
Perhaps this is too detailed, I think you can combine all the replies into one class file 👍

AwJoey commented on src/main/java/thecount/storage/Storage.java:
```diff
@@ -0,0 +1,142 @@
+        }
+    }
+
+    private void handleTaskDone(TaskList tasks, String isTaskDone, char index) {
```
I think a description for this method could be good!

AwJoey commented on src/main/java/thecount/storage/Storage.java:
```diff
@@ -0,0 +1,142 @@
+        }
+    }
+
+    private void processLine(String line, TaskList tasks) {
```
I think a description for this method could be good too!

AwJoey reviewed (commented):
Great work! 💯 Included some suggestions to consider too 😄

cliftonchee committed:
Add padding for dialog boxes

cliftonchee committed:
Change Elmo image

cliftonchee committed:
Fix Checkstyle

cliftonchee committed:
Add Ui.png

cliftonchee committed:
Update README.md

cliftonchee committed:
Update README.md

cliftonchee committed:
Update README.md

cliftonchee committed:
Update README.md

cliftonchee committed:
Fix build issues with other OS
